bindings/node.js/kzg.cxx

```diff
@@ -152,190 +152,215 @@ Napi::Value BlobToKzgCommitment(const Napi::CallbackInfo& info) {
   }
 
-  bool verification_result;
-  ret = verify_aggregate_kzg_proof(
-    &verification_result,
-    blobs,
-    commitments,
-    blobs_count,
+  auto blob_bytes = extract_byte_array_from_param(info, 0, "blob");
+  auto commitment_bytes = extract_byte_array_from_param(info, 1, "commitmentBytes");
+  auto proof_bytes = extract_byte_array_from_param(info, 2, "proofBytes");
+  auto kzg_settings = info[3].As<Napi::External<KZGSettings>>().Data();
+
+  if (env.IsExceptionPending()) {
+    return env.Null();
+  }
+
+  bool out;
+  C_KZG_RET ret = verify_blob_kzg_proof(
+    &out,
+    (Blob *)blob_bytes,
+    (Bytes48 *)commitment_bytes,
     (Bytes48 *)proof_bytes,
     kzg_settings
   );
 
-  free(commitments);
-  free(blobs);
-
   if (ret != C_KZG_OK) {
-    Napi::Error::New(
-      env,
-      "verify_aggregate_kzg_proof failed with error code: " + std::to_string(ret)
-    ).ThrowAsJavaScriptException();
+    Napi::TypeError::New(env, "Error in verifyBlobKzgProof").ThrowAsJavaScriptException();
     return env.Null();
   }
 
-  return Napi::Boolean::New(env, verification_result);
+  return Napi::Boolean::New(env, out);
 }
 
-// verifyKzgProof: (commitmentBytes: Bytes48, zBytes: Bytes32, yBytes: Bytes32, proofBytes: Bytes48, setupHandle: SetupHandle) => boolean;
-Napi::Value VerifyKzgProof(const Napi::CallbackInfo& info) {
+// verifyBlobKzgProofBatch: (blobs: Blob[], commitmentsBytes: Bytes48[], proofsBytes: Bytes48[], setupHandle: SetupHandle) => boolean;
+Napi::Value VerifyBlobKzgProofBatch(const Napi::CallbackInfo& info) {
   auto env = info.Env();
 
   size_t argument_count = info.Length();
-  size_t expected_argument_count = 5;
+  size_t expected_argument_count = 4;
   if (argument_count != expected_argument_count) {
     return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
   }
 
-  auto commitment_bytes = extract_byte_array_from_param(info, 0, "commitmentBytes");
-  auto z_bytes = extract_byte_array_from_param(info, 1, "zBytes");
-  auto y_bytes = extract_byte_array_from_param(info, 2, "yBytes");
-  auto proof_bytes = extract_byte_array_from_param(info, 3, "proofBytes");
-  auto kzg_settings = info[4].As<Napi::External<KZGSettings>>().Data();
+  auto blobs_param = info[0].As<Napi::Array>();
+  auto commitments_param = info[1].As<Napi::Array>();
+  auto proofs_param = info[2].As<Napi::Array>();
+  auto kzg_settings = info[3].As<Napi::External<KZGSettings>>().Data();
 
-  if (env.IsExceptionPending()) {
+  auto blobs_count = blobs_param.Length();
+  auto commitments_count = commitments_param.Length();
+  auto proofs_count = proofs_param.Length();
+
+  if (blobs_count != commitments_count || blobs_count != proofs_count) {
+    Napi::Error::New(env, "verifyBlobKzgProofBatch requires equal number of blobs/commitments/proofs")
+      .ThrowAsJavaScriptException();
     return env.Null();
   }
 
+  auto blobs = (Blob*)calloc(blobs_count, sizeof(Blob));
```

Comment:
Wow this required much more C code than all the other functions and actually most other bindings. Isn't there a simpler way to do things with Napi?

Reply:
Yeah that section is pretty ugly. I'd love for a nodejs person to improve it.

Reply:
Hello @dgcoffman. Can you please take a look at this section please? :)

Reply:
This is what the code to extract C-style byte arrays from Napi wrapped C++ data structures actually looks like. It's fine.

Reply:
Hmmm, isn't there a way to represent the arrays as byte arrays from the start (e.g. an [`ArrayBuffer`](https://github.com/nodejs/node-addon-api/blob/main/doc/array_buffer.md)), so that we don't need to do this entire `Array -> Uint8Array` conversion? It's quite error prone (as you can see in the various comments I left) and more complicated than other bindings.

Reply:
Let's look into this more later.